tests/unit/constraints/test_tabular.py

```diff
@@ -1274,3 +1274,174 @@ def test_reverse_transform(self):
+
+        # Assert
+        expected_out = pd.Series([True, True, True])
+        pd.testing.assert_series_equal(expected_out, out)
+
+    def test_is_valid_negative_digit_valid(self):
+        """Test the ``Rounding.is_valid`` method for a valid data.
+
+        If the data is within threshold of the desired rounding, the result
+        is a series of ``True`` values.
+
+        Input:
+        - Table data with desired decimal places (pandas.DataFrame)
+        Output:
+        - Series of ``True`` values (pandas.Series)
+        """
+        # Setup
+        columns = ['b']
+        digits = -2
+        tolerance = 1
+        instance = Rounding(columns=columns, digits=digits, tolerance=tolerance)
+
+        # Run
+        table_data = pd.DataFrame({
+            'a': [1, 2, 3],
+            'b': [401, 500, 6900],
+            'c': [5.313, 7.12, 9.120]
+        })
+        out = instance.is_valid(table_data)
+
+        # Assert
+        expected_out = pd.Series([True, True, True])
+        pd.testing.assert_series_equal(expected_out, out)
+
+    def test_is_valid_non_valid(self):
+        """Test the ``Rounding.is_valid`` method for a non-valid data.
+
+        If the data is not within threshold of the desired rounding, the
+        result is a series of ``False`` values.
+
+        Input:
+        - Table data not with the desired decimal places (pandas.DataFrame)
+        Output:
+        - Series of ``False`` values (pandas.Series)
+        """
+        # Setup
+        columns = ['b', 'c']
+        digits = 3
+        tolerance = 1e-4
+        instance = Rounding(columns=columns, digits=digits, tolerance=tolerance)
+
+        # Run
+        table_data = pd.DataFrame({
+            'a': [1, 2, 3],
+            'b': [4.1212, 5.5, 6.9017],
+            'c': [5.3, 7.12989, 9.0]
+        })
+        out = instance.is_valid(table_data)
+
+        # Assert
+        expected_out = pd.Series([False, False, False])
+        pd.testing.assert_series_equal(expected_out, out)
+
+    def test_is_valid_negative_digit_non_valid(self):
+        """Test the ``Rounding.is_valid`` method for a not valid data.
+
+        If the data is within threshold of the desired rounding, the result
+        is a series of ``True`` values.
+
+        Input:
+        - Table data with desired decimal places (pandas.DataFrame)
+        Output:
+        - Series of ``True`` values (pandas.Series)
+        """
+        # Setup
+        columns = ['b']
+        digits = -2
+        tolerance = 1
+        instance = Rounding(columns=columns, digits=digits, tolerance=tolerance)
+
+        # Run
+        table_data = pd.DataFrame({
+            'a': [1, 2, 3],
+            'b': [411, -551, 690050],
+            'c': [5.313, 7.12, 9.120]
+        })
+        out = instance.is_valid(table_data)
+
+        # Assert
+        expected_out = pd.Series([False, False, False])
+        pd.testing.assert_series_equal(expected_out, out)
+
+    def test_reverse_transform(self):
```

Comment:
We should probably test this multiple times two, with different digit levels (0, >0, <0), random Null values, and unaffected columns.